src/ops.c

```diff
@@ -249,23 +256,32 @@ shift_line(
 	}
 	else
 	    i += amount;
-	count = (long long)i * (long long)sw_val;
+	count = i * sw_val;
```

Comment:
Maybe something like (I don't think "`long = int*int` needs overflow check?)
```
#ifdef HAVE_NO_LONG_LONG
count = (long)i * (long)sw_val;
#else
count = (long long)i * (long long)sw_val;
#endif
```

Reply:
> I don't think "`long = int*int` needs overflow check?

On some platforms (e.g., Windows and 32-bit Linux) long and int are the same width.  A lot of Vim's code would benefit from using fixed-width integer types to avoid this inconsistency across platforms.

Reply:
> On some platforms (e.g., Windows and 32-bit Linux) long and int are the same width. 
Doh! Of course.

> A lot of Vim's code would benefit from using fixed-width integer types to avoid this inconsistency across platforms.

I don't disagree. But then some stuff, like uint64_t, doesn't map 32bit VAX?

Reply:
> I don't disagree. But then some stuff, like uint64_t, doesn't map 32bit VAX?

Yes, you are right @errael . On 32bit architectures - including VAX - this needs to be handled. This is part of the porting acrobatics between different architectures. 
This is exactly what I did.

I wonder, why this "long long overflow" related conversation is started here. This piece of code has been already there. How did it pass the CI tests when it was first committed - and all other consecutive runs since?

Reply:
> I wonder, why this "long long overflow" related conversation is started here.

This change takes away the `long long` casting. Is that intentional? (that's how the conversation got started here).

Reply:
And  I wanted to point out how multiply overflow might be detected, if that's needed for VAX32.

Reply:
Thank you for turning my attention to this.
I did not intentionally remove the casting there. It was a consequence or working on such (four different architectures) without functional git. The mail has changed since my checkout... and I worked with 9..0.2112 and copied over the changed files.

I have corrected the mistake... please review.
Thanks.